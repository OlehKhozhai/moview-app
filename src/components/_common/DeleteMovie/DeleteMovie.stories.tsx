import React from 'react';
import { Meta } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import DeleteMovie from './index';

export default {
  component: DeleteMovie,
  title: 'components/DeleteMovie',
} as Meta;

const props = () => ({
  onConfirm: action('onConfirm'),
});

export const DeleteMovieComponent = () => (
  <div style={{ width: 400, textAlign: 'left' }}>
    <DeleteMovie {...props()} />
  </div>
);
