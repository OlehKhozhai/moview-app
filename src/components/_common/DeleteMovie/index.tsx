import React from 'react';

import Button from 'components/_common/Button';
import styles from './styles.module.scss';

type DeleteMovieProps = {
  onConfirm: () => void;
};

const DeleteMovie: React.FC<DeleteMovieProps> = ({ onConfirm }) => {
  return (
    <>
      <h2 className={styles.title}>Delete movie</h2>
      <h5 className={styles.subTitle}>Are you sure you want to delete this movie?</h5>
      <Button title="Confirm" className={styles.button} onClick={onConfirm} />
    </>
  );
};

export default React.memo(DeleteMovie);
