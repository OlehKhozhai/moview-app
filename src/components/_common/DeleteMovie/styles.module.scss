@import 'assets/styles/mixins/colors';

.title {
  text-transform: uppercase;
  font-size: 40px;
  text-align: left;
}

.subTitle {
  margin: 50px 0;
  font-size: 18px;
}

.button {
  margin-left: auto;
  display: block;
}
