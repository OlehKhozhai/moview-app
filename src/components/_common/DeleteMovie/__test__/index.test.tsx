import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import DeleteMovie from '../index';

describe('DeleteMovie component', () => {
  test('should render DeleteMovie component by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <DeleteMovie onConfirm={() => undefined} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should trigger handleOnConfirm method', () => {
    const handleOnConfirm = jest.fn();
    const component = create(<DeleteMovie onConfirm={handleOnConfirm} />);

    component.root.props.onConfirm();

    expect(handleOnConfirm).toHaveBeenCalled();
  });
});
