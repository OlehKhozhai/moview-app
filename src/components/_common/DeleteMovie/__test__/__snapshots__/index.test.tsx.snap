// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovie component should render DeleteMovie component by default 1`] = `
Array [
  <h2
    className="title"
  >
    Delete movie
  </h2>,
  <h5
    className="subTitle"
  >
    Are you sure you want to delete this movie?
  </h5>,
  <button
    className="root primary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Confirm
  </button>,
]
`;
